src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/multitable/MultitableFlowsSpec.groovy

```diff
@@ -136,237 +135,188 @@ mode with existing flows and hold flows of different table-mode types"() {
-
-        then: "Flow rules on the switch are recreated in multi table mode"
-        def flowInfoFromDb3
+        then: "Flow rules go to multi table mode"
         wait(RULES_INSTALLATION_TIME) {
-            flowInfoFromDb3 = database.getFlow(flow.flowId)
-            verifyAll(northbound.getSwitchRules(sw.dpId).flowEntries) { rules ->
-                rules.find { it.cookie == flowInfoFromDb3.forwardPath.cookie.value }.tableId == INGRESS_RULE_MULTI_TABLE_ID
-                rules.find { it.cookie == flowInfoFromDb3.reversePath.cookie.value }.tableId == INGRESS_RULE_MULTI_TABLE_ID
-                rules.findAll { it.cookie in sharedRules*.cookie }*.tableId.unique() == [SHARED_RULE_TABLE_ID]
-            }
+            "flow rules are in tables"(sw, flow) == [INGRESS_RULE_MULTI_TABLE_ID]
         }
+        cleanup:
+        flowHelperV2.deleteFlow(flow.flowId)
+        initSwProps && revertSwitchToInitState(sw, initSwProps)
 
-        when: "Update switch properties(multi_table: false) on the switch"
-        northbound.updateSwitchProperties(sw.dpId, changeSwitchPropsMultiTableValue(initSwProps, false))
+        where:
+        eventName           | event
+        /* https://github.com/telstra/open-kilda/issues/4865
+        "synchronize flow"  | {FlowRequestV2 testFlow ->
+            with(northbound.synchronizeFlow(testFlow.flowId)) {!it.rerouted}} */
+        "update flow"       | {FlowRequestV2 testFlow ->
+            flowHelperV2.updateFlow(testFlow.flowId,
+                    testFlow.tap { it.description = it.description + " updated" })}
+    }
 
-        and: "Reroute(intentional) the flow via APIv1"
-        with(northbound.rerouteFlow(flow.flowId)) { !it.rerouted }
-        wait(rerouteDelay + WAIT_OFFSET) {
-            assert northboundV2.getFlowStatus(flow.flowId).status == FlowState.UP
-        }
+    @Tidy
+    @Tags([LOW_PRIORITY])
+    def "Single-switch flow rules are not updated when try to reroute the flow"() {
+        given: "An active switch"
+        def sw = topology.activeSwitches.find { it.features.contains(SwitchFeature.MULTI_TABLE) }
+        SwitchPropertiesDto initSwProps = switchHelper.getCachedSwProps(sw.dpId)
 
-        then: "Flow rules on the switch are not recreated in single table mode because flow wasn't rerouted"
-        verifyAll(northbound.getSwitchRules(sw.dpId).flowEntries) { rules ->
-            rules.find { it.cookie == flowInfoFromDb3.forwardPath.cookie.value }.tableId == INGRESS_RULE_MULTI_TABLE_ID
-            rules.find { it.cookie == flowInfoFromDb3.reversePath.cookie.value }.tableId == INGRESS_RULE_MULTI_TABLE_ID
-            rules.findAll { it.cookie in sharedRules*.cookie }*.tableId.unique() == [SHARED_RULE_TABLE_ID]
+        and: "Multi table mode is disabled on it"
+        if (initSwProps.multiTable) {
+            northbound.updateSwitchProperties(sw.dpId, changeSwitchPropsMultiTableValue(initSwProps, false))
```

Comment:
you changed test case. Prev test checked multi -> single changed. This tests single -> multi. It this particular case it ok, because reroute must not change mode in any case. But As you named PR as "refactoring" you shouldn't change case coverage

Reply:
We have checks on 'multi -> single' switch transition in other tests, e.g. `"Flow rules are re(installed) according to switch props while syncing switch and rules"`. So while the coverage of this test is decreased, overall tests coverage stays effectively the same.